src/server/utils/redirectAllowList.js

```diff
@@ -0,0 +1,35 @@
+import wcmatch from 'wildcard-match';
+
+let redirectAllowList = [];
+
+export const setRedirectAllowList = (allowList) => {
+  if (!Array.isArray(allowList)) {
+    console.error('redirectAllowList is not an array');
+    return;
+  }
+  redirectAllowList = [];
+  allowList.forEach((url) => {
+    if (url.includes('http://')) {
+      console.error(`Insecure protocols (http://) are not allowed to be redirect locations. Ignoring '${url}' listed in redirectAlowList configuration.`);
+      return;
+    }
+    if (!url.includes('https://')) {
+      const modifiedUrl = 'https://'.concat(url);
+      redirectAllowList.push(modifiedUrl);
+      return;
+    }
+    redirectAllowList.push(url);
+  });
+};
+
+export const getRedirectAllowList = () => redirectAllowList;
```

Comment:
give out a copy so it cant be mutated?

Reply:
done

Reply:
<img width="458" alt="image" src="https://github.com/americanexpress/one-app/assets/2575450/f14b6052-f7a3-476f-926e-b8d3d6392d9e">
did you?